src/types/media.ts

```diff
@@ -33,3 +57,35 @@ export const isGif = (mimeType?: MimeEnum) => {
 
   return mimeType === MimeEnum.gif;
 };
+
+// perform HEAD request on IPFS file CID to resolve `mimeType`, along with `isAudio`, `isVideo`, `isImage` flags
+export const updateMimeTypes = async (ipfsFile: IPFSFile) => {
+  const ipfsHash = ipfsFile.cid;
+  let response: any;
+  let errorMsg: string | undefined = undefined;
+  let contentType: any = '';
+
+  try {
+    response = await axios.head(`${process.env.IPFS_ENDPOINT}${ipfsHash}`, { timeout: parseInt(process.env.METADATA_REQUEST_TIMEOUT!) })
+    contentType = response.headers['content-type']?.toLowerCase();
+  } catch (e: any) {
+    errorMsg = `Error: failed to fetch mime type for ipfs hash: ${ipfsHash} with error: ${e.message}`;
+  }
+
+  if (contentType && !Object.values(MimeEnum).includes(contentType)) {
+    errorMsg = `Error: unsupported mime type '${contentType}' for ipfs hash: ${ipfsHash}`;
+  }
+
+  const updated = ipfsFile;
```

Comment:
Let's not change the object that's passed in as a parameter, since then this loses it's purity:

```suggestion
  const updated = { ...ipfsFile };
```